rdt/hyper_transformer.py

```diff
@@ -320,6 +324,40 @@ def get_transformer_tree_yaml(self):
 
         return yaml.safe_dump(dict(modified_tree))
 
+    def detect_initial_config(self, data):
+        """Print the configuration of the data.
+
+        This method detects the ``sdtype`` and transformer of each field in the data
+        and then prints them as a json object.
+
+        NOTE: This method partially resets the state of the ``HyperTransformer``.
+        Previously set ``sdtypes`` or transformers will be lost.
+
+        Args:
+            data (pd.DataFrame):
+                Data which will have its configuration detected.
+        """
+        # Reset the state of the HyperTransformer
+        self.field_data_types = {}
+        self.field_transformers = {}
+
+        # Set the sdtypes and transformers of all fields to their defaults
+        for field in data:
+            self._set_field_data_type(data, field)
+            field_sdtype = self.field_data_types[field]
+            self.field_transformers[field] = get_default_transformer(field_sdtype)
+
+        print('Detecting a new config from the data ... SUCCESS')  # noqa: T001
+        print('Setting the new config ... SUCCESS')  # noqa: T001
+
+        config = {
+            'sdtypes': self.field_data_types,
+            'transformers': {k: repr(v) for k, v in self.field_transformers.items()}
+        }
+
+        print('Config:')  # noqa: T001
+        print(json.dumps(config, indent=4))  # noqa: T001
```

Comment:
should we print as json or yaml? @npatki

Reply:
@fealho nice print ! 😸 I didn't know you could do this! I usually used `preatyprint`  but this is a nice hack! 👍🏻